runtime/src/accounts.rs

```diff
@@ -1805,4 +1837,36 @@ mod tests {
         accounts.squash(1);
         assert_eq!(accounts.transaction_count(1), 2);
     }
+
+    #[test]
+    fn test_load_by_program() {
+        let paths = get_tmp_accounts_path!();
+        let accounts_db = AccountsDB::new(0, &paths.paths);
+
+        // Load accounts owned by various programs into AccountsDB
+        let pubkey0 = Keypair::new().pubkey();
+        let account0 = Account::new(1, 0, Pubkey::new(&[2; 32]));
```

Comment:
account0 and account1 have the same pubkey, that’s ok?

Reply:
They are specifically owned by the same program id, yeah. To test whether the methods actually return multiples.